src/os/ObjectStore.h

```diff
@@ -2315,6 +2315,7 @@ class ObjectStore {
   * - num objects - total (including witeouts) object count to measure used space for.
   */
   virtual uint64_t estimate_objects_overhead(uint64_t num_objects) = 0;
+  virtual int apply_layout_settings(coll_t cid) { assert(0); return 0; }
```

Comment:
IIUC, only FileStore needs to tackle this split-merge problem. can we `dynamic_cast<>` the `ObjectStore` in `ceph_objectstore_tool.cc`, if it is actually a `FileStore`, then we can `apply_layout_settings()` to it.